utils/dataHandling.ts

```diff
@@ -0,0 +1,277 @@
+import { load } from '@welshman/net';
+import { deriveProfile } from '@welshman/app';
+import { Router } from '@welshman/router';
+import { publishThunk } from '@welshman/app';
+import { makeEvent } from '@welshman/util';
+import { pubkey } from '@welshman/app';
+
+/**
+ * Establishes initial relay connections on app launch
+ * @returns Promise that resolves when connections are established
+ */
+export const establishRelayConnections = async (): Promise<void> => {
+  const startTime = Date.now();
+  console.log("Establishing relay connections on app launch...");
+  
+  try {
+    const defaultRelays = ["wss://relay.damus.io/", "wss://nos.lol/"];
+    console.log("Using default relays:", defaultRelays);
+    
+    const events = await load({ 
+      relays: defaultRelays, 
+      filters: [{ kinds: [0], limit: 1 }] //load 1 profile event to establish connections
+    });
+    
+    const duration = Date.now() - startTime;
+    console.log(`Relay connections established in ${duration}ms, loaded ${events.length} events`);
+  } catch (error) {
+    const duration = Date.now() - startTime;
+    console.warn(`Failed to establish relay connections after ${duration}ms:`, error);
+  }
+};
+
+/**
+ * Loads a user's profile from Nostr relays
+ * @param pubkey - The user's public key
+ * @returns Promise that resolves with the profile data
+ */
+export const loadUserProfile = async (pubkey: string): Promise<any> => {
+  if (!pubkey) {
+    throw new Error('Pubkey is required');
+  }
+
+  try {
+    console.log('[PROFILE] Loading profile for pubkey:', pubkey);
+    
+    const router = Router.get();
+    let readRelays = router.ForPubkey(pubkey).getUrls();
```

Comment:
This is actually backwards, ForPubkey is used to select relay the user _reads notes created by other people that mention him from_. You want `FromPubkey`, which is the relays the user writes his own notes to. Since profile is created by the user, that's where it will go.